HumanDynamicsEstimationLibrary/algorithms/src/DynamicalInverseKinematics.cpp

```diff
@@ -0,0 +1,1378 @@
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setPositionFeedbackGain(positionFeedbackGain);
+    target->second.setOrientationFeedbackGain(orientationFeedbackGain);
+    target->second.setLinearVelocityFeedforwardGain(linearVelocityFeedforwardGain);
+    target->second.setAngularVelocityFeedforwardGain(angularVelocityFeedforwardGain);
+    return true;
+}
+
+bool DynamicalInverseKinematics::updateTargetWeights(const std::string& linkName,
+                                                   const double linearVelocityWeight,
+                                                   const double angularVelocityWeight)
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setLinearVelocityWeight(linearVelocityWeight);
+    target->second.setAngularVelocityWeight(angularVelocityWeight);
+    return true;
+}
+
+
+
+bool DynamicalInverseKinematics::getJointsConfigurationSolution(iDynTree::VectorDynSize& jointsConfiguration) const
+{
+    if (jointsConfiguration.size() == pImpl->m_state.s.size()) {
+        jointsConfiguration = pImpl->m_state.s;
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+bool DynamicalInverseKinematics::getBasePoseSolution(iDynTree::Transform& baseTransform) const
+{
+    baseTransform = iDynTree::Transform(pImpl->m_state.W_R_B, iDynTree::Position(pImpl->m_state.W_p_B));
+    return true;
+}
+
+bool DynamicalInverseKinematics::getBasePoseSolution(iDynTree::Vector3& basePosition, iDynTree::Rotation& baseRotation) const
+{
+    basePosition = pImpl->m_state.W_p_B;
+    baseRotation = pImpl->m_state.W_R_B;
+    return true;
+}
+
+bool DynamicalInverseKinematics::getConfigurationSolution(iDynTree::Transform& baseTransform,
+                                                          iDynTree::VectorDynSize& jointsConfiguration) const
+{
+    return getJointsConfigurationSolution(jointsConfiguration) && getBasePoseSolution(baseTransform);
+}
+bool DynamicalInverseKinematics::getConfigurationSolution(iDynTree::Vector3& basePosition,
+                                                          iDynTree::Rotation& baseRotation,
+                                                          iDynTree::VectorDynSize& jointsConfiguration) const
+{
+    return getJointsConfigurationSolution(jointsConfiguration) && getBasePoseSolution(basePosition, baseRotation);
+}
+
+bool DynamicalInverseKinematics::getVelocitySolution(iDynTree::Twist& baseVelocity,
+                                                     iDynTree::VectorDynSize& jointsVelocity) const
+{
+    getBaseVelocitySolution(baseVelocity);
+    getJointsVelocitySolution(jointsVelocity);
+
+    return true;
```

Comment:
probably, similar to the previous function:

```
return     getBaseVelocitySolution(baseVelocity)  && getJointsVelocitySolution(jointsVelocity);
```

Reply:
addressed in https://github.com/robotology/human-dynamics-estimation/pull/265/commits/464177c5d794cb35c8739bdf76a23de82ed1cc2d